packages/schematics/src/command-line/shared.ts

```diff
@@ -18,19 +18,44 @@ import { statSync } from 'fs';
 import * as appRoot from 'app-root-path';
 import { readJsonFile } from '../utils/fileutils';
 import { YargsAffectedOptions } from './affected';
+const ignore = require('ignore');
 
-export type ImplicitDependencyEntry = { [key: string]: string[] };
+export type ImplicitDependencyEntry = { [key: string]: '*' | string[] };
+export type NormalizedImplicitDependencyEntry = { [key: string]: string[] };
 export type ImplicitDependencies = {
-  files: ImplicitDependencyEntry;
-  projects: ImplicitDependencyEntry;
+  files: NormalizedImplicitDependencyEntry;
+  projects: NormalizedImplicitDependencyEntry;
 };
 
+export interface NxJson {
+  exclude?: string[];
+  implicitDependencies?: ImplicitDependencyEntry;
+  npmScope: string;
+  projects: {
+    [projectName: string]: NxJsonProjectConfig;
+  };
+}
+
+export interface NxJsonProjectConfig {
+  implicitDependencies?: string[];
+  tags?: string[];
+}
+
+function readFileIfExisting(path: string) {
+  return fs.existsSync(path) ? fs.readFileSync(path, 'UTF-8').toString() : '';
+}
+
+const ig = ignore();
+ig
+  .add(readFileIfExisting(appRoot.resolve('.gitignore')))
```

Comment:
We don't use `resolve` anywhere else here. For instance, this is how we read nx.json:
```
  const config = readJsonFile(`${appRoot.path}/nx.json`);
```

Could you make it consistent?